Target net11.0 / .NET 11

ViktorHofer opened the pull request:
Contributes to https://github.com/dotnet/runtime/issues/118583

For reference, this was last year's PR for net10.0.  https://github.com/dotnet/runtime/pull/106599

ViktorHofer committed:
Branding to 11.0.0

ViktorHofer committed:
Target net11.0

ViktorHofer commented:
Not necessarily but it would be better to get that in first. I included the branding update in this PR (see eng/Versions.props changes).

ericstj committed:
Temporarily use net10.0 pruning data on net11.0

ericstj committed:
Hardcode version for intellisense files

ericstj committed:
Make WasmSymbolicator target NetCoreAppMinimum

ericstj committed:
Fixing more exclusive usage of NetCoreAppPrevious

ericstj committed:
Fix a few APICompat issues after targeting net11.0

ViktorHofer committed:
Remove DisablePackageBaselineValidation from newly 10.0 shipped packages

ViktorHofer committed:
Set NETCoreAppMaximumVersion

ViktorHofer committed:
Merge branch 'main' into net11

ViktorHofer committed:
Delete tensosrs net8.0 test project

lewing committed:
Use 10.0.100 band version until we update to an 11 sdk

am11 commented:
crossgen2, ilc and cdac are published as AOT. One difference I've found in cDAC projects is missing these props:

```csproj
    <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
    <AppendTargetFrameworkToOutputPath Condition="'$(BuildingInsideVisualStudio)' == 'true'">true</AppendTargetFrameworkToOutputPath>
```

AppendRuntimeIdentifierToOutputPath is in ilc, but not crossgen2 so it might be optional.

ericstj committed:
Use net11 pruning data workaround in trimming tests

lewing committed:
Rework workload manifest versions now that they are only used for testing

lewing committed:
Remove added blank lines

lewing committed:
Check current framework version and fix conditions

lewing committed:
Add net10 manifests

lewing committed:
Parse global.json to get the base sdk version if NETCoreSdkVersion is not set

lewing committed:
Make the regex parsing json a little more robust

lewing committed:
Fix the localized files

lewing committed:
Remove unused darc dependency on transport manifest package

lewing committed:
Fix sdkbandversion computation

lewing committed:
Make WasmNativeWorkloadAvailable follow the same pattern for net10/11

ViktorHofer commented on eng/Versions.props:
```diff
@@ -146,25 +154,21 @@
-    <!-- workloads -->
+
+    <!-- emscripten workload package when testing workloads -->
+    <MicrosoftNETRuntimeEmscriptenVersion>$(MicrosoftDotNetApiCompatTaskPackageVersion)</MicrosoftNETRuntimeEmscriptenVersion>
```
Note that reading from another property in Versions.props is an anti-pattern as this will have the wrong value when building from source. That overrides versions after the Versions.props import. We usually put things that read from Version.Details.props / Versions.props into Directory.Build.props.

Might not matter here if this path isn't exercised in source-build but wanted to mention it.

See https://github.com/dotnet/sdk/blob/0d31ba892465268a030b3563df0bddef39855f6e/Directory.Build.props#L97 for an example

ViktorHofer replied:
Actually, this doesn't just apply to source-build but also to the repo build inside the VMR. So it might matter here. Not sure.

steveisok replied:
To clarify - if it's shipped out of source build. My understanding may be stale, but I didn't think we did that for wasm.

lewing replied:
https://github.com/dotnet/runtime/issues/122354

lewing committed:
Add back transport version and use it for the sdk version computation

ericstj committed:
Update test resources with latest assembly version

sbomer committed:
Fix ILLink.RoslynAnalyzer.Tests

ericstj commented:
> The linker test assemblies are built against NetCoreAppToolCurrent which is 10.0. Then during test execution, we recompile individual test files against the live libraries, along with a reference to one of the already-built linker test assemblies. I think this is safe to suppress - I'll prepare a fix.

Thank you @sbomer - you may want to also include CS1702 as well - looks like that is for build and revision.  CS1701 is for major and minor (which is all we need here).

ericstj commented:
@lewing  - that error seems to originate from here -- https://github.com/dotnet/runtime/blob/a0d68590430b83e7b5de292a84e68a20169faf2a/src/tasks/WasmAppBuilder/mono/ManagedToNativeGenerator.cs#L78

My guess is that the `Assemblies` parameter is missing a path to corelib.  [Look further upstream](https://artprodcus3.artifacts.visualstudio.com/A6fcc92e5-73a7-4f88-8d13-d9045b45fb27/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy1wdWJsaWMvcHJvamVjdElkL2NiYjE4MjYxLWM0OGYtNGFiYi04NjUxLThjZGNiNTQ3NDY0OS9idWlsZElkLzEyMjA4MjkvYXJ0aWZhY3ROYW1lL0xvZ3NfQnVpbGRfQXR0ZW1wdDFfYnJvd3Nlcl93YXNtX3dpbmRvd3NfUmVsZWFzZV9MaWJyYXJ5VGVzdHM1/content?format=file&subPath=%2FRelease%2FBuild.binlog) to see where that comes from.

steveisok committed:
Don't append tfm to output path for mscordaccore_universal

steveisok commented:
> `<AppendTargetFrameworkToOutputPath>true` does make a difference:

@am11 good suggestion (as usual)

ericstj commented:
> My guess is that the `Assemblies` parameter is missing a path to corelib. [Look further upstream](https://artprodcus3.artifacts.visualstudio.com/A6fcc92e5-73a7-4f88-8d13-d9045b45fb27/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy1wdWJsaWMvcHJvamVjdElkL2NiYjE4MjYxLWM0OGYtNGFiYi04NjUxLThjZGNiNTQ3NDY0OS9idWlsZElkLzEyMjA4MjkvYXJ0aWZhY3ROYW1lL0xvZ3NfQnVpbGRfQXR0ZW1wdDFfYnJvd3Nlcl93YXNtX3dpbmRvd3NfUmVsZWFzZV9MaWJyYXJ5VGVzdHM1/content?format=file&subPath=%2FRelease%2FBuild.binlog) to see where that comes from.

It looks to me like something in https://github.com/dotnet/runtime/blob/main/src/mono/wasm/build/WasmApp.Common.targets is failing to publish the runtime bits.  Was talking to @steveisok offline and it could be due to no runtime pack being available.  It looks like in 10.0 we had 10.0 runtime packs available by the time we were retargeting but we don't have that right now.  I'm not certain if this is the problem, but it seems plausible.

sbomer committed:
Also CS1702

ericstj committed:
Use AnyHaveMetadataValue rather than comparing to empty

Workaround https://github.com/dotnet/msbuild/issues/12877

lewing committed:
Merge branch 'main' into net11

ericstj committed:
Fix pruning error in installer tests

ericstj commented:
## /native/managed/cdac/ --> ` Unable to find package Microsoft.AspNetCore.App.Runtime.linux-riscv64. No packages exist with this id in source(s)` - @steveisok 

Hitting this for linux-riscv64, linux-loongarch64, and freebsd-x64.  They aren't connected to [live runtime packs](https://github.com/dotnet/runtime/blob/2c911651ab05a78ca58f02777b34302ae3a2e07e/eng/targetingpacks.targets#L69).  Frankly I don't understand why these components need runtime packs at all since they don't appear to be applications.   I discussed with @steveisok and he will follow up to get understanding.

## WasmBuildTests --> `You must install or update .NET to run this application` - @lewing

The helix tests are provisioning an older runtime (RC1) than the test assembly was built with (RC2).  Tests could be updated to install a matching one.  Or we could wait for the GA SDK to land, in which case we'd be depending on the same runtime for both.

am11 commented:
> Hitting this for linux-riscv64, linux-loongarch64, and freebsd-x64. They aren't connected to [live runtime packs](https://github.com/dotnet/runtime/blob/2c911651ab05a78ca58f02777b34302ae3a2e07e/eng/targetingpacks.targets#L69).

They are the community supported platforms using bootstrapping mechanism (aka from-scratch build): https://github.com/dotnet/runtime/blob/3e44f568cb24aa68f863d925b67fa5408b734d54/docs/workflow/building/coreclr/cross-building.md#building-coreclr-with-bootstrapping. First pass builds minimal subsets: shared framework and tools, second pass moves them in `artifacts/bootstrap` and uses that path to find apphost, nativeaot/crossgen2 runtime packs, RID graph etc. for the full build. So they use a different type of live packs.

> Frankly I don't understand why these components need runtime packs at all since they don't appear to be applications. I discussed with @steveisok and he will follow up to get understanding.

They need it to publish packages with executables during the build like Microsoft supported platforms.

am11 commented:
Yes, all tools which are in linux-arm64 SDK are included in SDKs for those RIDs, e.g. https://github.com/filipnavara/dotnet-riscv/releases/tag/10.0.100.

lewing committed:
Install more bits

lewing committed:
Do more workload tests

lewing committed:
Add missing project

lewing committed:
add bootstrap patch

steveisok commented:
> both passed the restore and cDAC package build and later failing at ApiCompat stage.

I tried it both locally and I see @lewing pushed a change. Neither work. 

The `TargetFramework` part of the change produces `Ahead-of-time compilation is not supported for the target runtime identifier 'freebsd-x64'`. That is a different message, but it's still tanking on restore.

lewing committed:
Merge branch 'main' into net11

am11 commented:
@lewing, it turned out my branch was old, was based on older commit (which is why appcompat issue was present in my runs). Could you please apply:

https://github.com/am11/runtime/commit/1dae2e447938643be1b80a194946a486a90989e2

it only affects community platforms and fixes the issue.

lewing committed:
Merge branch 'main' into net11

steveisok committed:
Make sure cdac uses local runtime packs

jkoritzinsky commented:
We should update the logic to always use the local packs when UseBootstrapLayout is true. Then it should do the "right thing".

lewing committed:
Temporarily satisfy emscripten net10 sdk with current

issue to track the work is https://github.com/dotnet/runtime/issues/122264

lewing committed:
Bump wbts targetframework

lewing committed:
Update src/native/managed/cdac/mscordaccore_universal/mscordaccore_universal.csproj

Co-authored-by: Adeel Mujahid <3840695+am11@users.noreply.github.com>

lewing committed:
Undo reading version from environment because it breaks in major version upgrades

It is tempting to read the version from an environment variable, but this
breaks scenarios where we do a major version upgrade (e.g., net10.0
to net11.0) because we may be running the test build from an older runtime.

lewing committed:
Merge branch 'net11' of https://github.com/dotnet/runtime into net11

am11 commented on src/mono/wasm/Wasm.Build.Tests/NonWasmTemplateBuildTests.cs:
```diff
@@ -25,9 +25,10 @@ public NonWasmTemplateBuildTests(ITestOutputHelper output, SharedBuildPerTestCla
-    private static readonly string s_latestTargetFramework = $"net{Environment.Version.Major}.0";
-    private static readonly string s_previousTargetFramework = $"net{Environment.Version.Major - 1}.0";
+    // are really the previous version.
+    private static readonly int TargetMajorVersion = 11; /* net11 */
```
`private const int`?

lewing committed:
More fixes

lewing committed:
Add placeholder for next bump

lewing committed:
bump the output dir

lewing committed:
Merge branch 'main' into net11

lewing committed:
Try something

lewing committed:
Force framework for blazor template app

lewing committed:
Try something else

lewing committed:
Handle both cases

lewing committed:
🙈

pavelsavara committed:
active isseueshttps://github.com/dotnet/runtime/issues/122338

kotlarmilos committed:
Use Apple mobile scouting queues

kotlarmilos committed:
Revert changes

ericstj commented on src/libraries/System.Resources.Extensions/tests/TestData.resources:
No, this is a binary file that is checked in.  I updated it.
